packages/ui-extensions/docs/surfaces/point-of-sale/reference/examples/scanner-api/conditional-scanner-example.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  CameraScanner,
+  Navigator,
+  Screen,
+  Stack,
+  Text,
+  extension,
+} from '@shopify/ui-extensions/point-of-sale';
+
+export default extension('pos.home.modal.render', (root, api) => {
+  api.scanner.scannerDataSubscribable.subscribe((data, source) => {
+    dataText.updateProps({
+      text: `Scanned data: ${data || ''}`,
+    });
+    sourceText.updateProps({
+      text: `Scanned data source: ${source || ''}`,
+    });
+  });
+
+  api.scanner.scannerSourcesSubscribable.subscribe((sources) => {
+    // Clear previous children to avoid duplicate appending
+    stack1.children = [];
+    stack2.children = [];
```

Comment:
`stack1` and `stack2` aren't defined yet here, so it's weird to read about them. Maybe this function should go beneath the definitions.